### ISSUE TITLE
Question: Will this plugin get updated to support API 24+? Because using this we get Error: android.os.FileUriExposedException

### ISSUE THREAD

**Issue by AmithRamesh** [rafear/nativescript-handle-file#1]
@rafear Please guide if any alternative or workaround solution. I don't want to target API 23 anymore.
Thank you.

**Comment by rafear** [comment]
Hi @AmithRamesh, I pretend upgrade to 24+ soon. But It will take a while before I get some time to modify.

**Comment by AmithRamesh** [comment]
Hi @rafear 
I changed code in handle-file.android.ts as shown below. And followed the FileProvider setup as mentioned [here](https://developer.android.com/reference/android/support/v4/content/FileProvider#ServeUri).
This is working for me, so hope this helps.

Changed the following lines of code --
` let intent = new android.content.Intent(android.content.Intent.ACTION_VIEW);
let mimeType = this.findExtension(extension);
intent.setDataAndType(android.net.Uri.fromFile(new java.io.File(file.path)), mimeType);
application.android.currentContext.startActivity(android.content.Intent.createChooser(intent, title));
`

to --

`let intent = new android.content.Intent(android.content.Intent.ACTION_VIEW);
let context = application.android.currentContext;
let mimeType = this.findExtension(extension);

// Set the file content URI using Android native APIs 
 intent.setDataAndType(android.support.v4.content.FileProvider.getUriForFile(context, context.getApplicationContext().getPackageName() + ".fileprovider", new java.io.File(file.path)), mimeType);
 // Grant permission for third party apps to read the content of this file 
intent.setFlags(android.content.Intent.FLAG_GRANT_READ_URI_PERMISSION);
context.startActivity(android.content.Intent.createChooser(intent, title));`

**Comment by rafear** [comment]
@AmithRamesh I'm going to change this today, if you wanna make a pull request to contribute with project feel free to do it, or in some hours I will change this lines.

Thanks for the help

**Comment by andreaCremisini** [comment]
@AmithRamesh I try to change my code with your lines, but it's not working. Do you change only this line or you create another file in xml (for example xml/provider_paths or Android Manifest)

**Comment by AmithRamesh** [comment]
@andreaCremisini Yeah you need to add an xml file(xml/provider_paths) as mentioned in the guide [here](https://developer.android.com/reference/android/support/v4/content/FileProvider#ServeUri).
And make sure you add `<provider>...</provider>` in your AndroidManifest with required info.

@rafear hey, sorry got caught up with something, was unable to get back to you. Looking forward to the update.

**Comment by andreaCremisini** [comment]
Yes, but I have some problem with tag "android:authorities=“com.mydomain.fileprovider” … There is an error if I use .fileprovider:

> Manifest merger failed : Attribute provider#android.support.v4.content.FileProvider@authorities value=(com.***.fileprovider) from AndroidManifest.xml:87:13-64
        is also present at [:nativescript_camera:] AndroidManifest.xml:14:13-60 value=(com.***.provider). If I use .provider, I don’t have an error, but not works

**Comment by AmithRamesh** [comment]
@andreaCremisini try this
`<provider
			android:name="android.support.v4.content.FileProvider"
			android:authorities="${applicationId}.fileprovider"
			android:exported="false"
			android:grantUriPermissions="true">
			<meta-data
				android:name="android.support.FILE_PROVIDER_PATHS"
				android:resource="@xml/provider_paths" />
		</provider>`

**Comment by andreaCremisini** [comment]
I tried your last message (i tried this a some days ago), but i have this error again:
JS: Error in openFile:  Error: java.lang.IllegalArgumentException: Failed to find configured root that contains /data/data/com.***.****/files/document.pdf
JS:     android.support.v4.content.FileProvider$SimplePathStrategy.getUriForFile(FileProvider.java:738)
JS:     android.support.v4.content.FileProvider.getUriForFile(FileProvider.java:417)
JS:     com.tns.Runtime.callJSMethodNative(Native Method)
JS:     com.tns.Runtime.dispatchCallJSMethodNative(Runtime.java:1101)
JS:     com.tns.Runtime.callJSMethodImpl(Runtime.java:983)
JS:     com.tns.Runtime.callJSMethod(Runtime.java:970)
JS:     com.tns.Runtime.callJSMethod(Runtime.java:954)
JS:     com.tns.Runtime.callJSMethod(Runtime.java:946)
JS:     com.tns.gen.android.view.GestureDetector_SimpleOnGestureListener_vendor_1_842640_TapAndDoubleTapGestureListenerImpl.onSingleTapUp(GestureDetector_SimpleOnGestureListener_vendor_1_842640_TapAndDoubleTapGestureListenerImpl.java:17)
JS:     android.view.GestureDetector.onTouchEvent(GestureDetector.java:640)
JS:     android.support.v4.view.Gesture...


My provider_paths file is: 
<?xml version="1.0" encoding="utf-8"?>
<paths xmlns:android="http://schemas.android.com/apk/res/android">
       <external-files-path
            name="external_files"
            path="." />

</paths>

**Comment by andreaCremisini** [comment]
My provider_paths file is: 
```
<?xml version="1.0" encoding="utf-8"?>
<paths xmlns:android="http://schemas.android.com/apk/res/android">
       <external-files-path
            name="external_files"
            path="." />

</paths>
```

**Comment by awebi** [comment]
@AmithRamesh Please what  is the correct providers_path to use. Thanks.

**Comment by AmithRamesh** [comment]
@awebi 

Add this to your Android manifest

<provider
	android:name="android.support.v4.content.FileProvider"
	android:authorities="${applicationId}.fileprovider"
	android:exported="false"
	android:grantUriPermissions="true">
	<meta-data
		android:name="android.support.FILE_PROVIDER_PATHS"
		android:resource="@xml/file_paths" />
</provider>  

And then create a file named file_paths in XML folder (./App_Resources/Android/xml/file_paths.xml)

<?xml version="1.0" encoding="utf-8"?>
<paths xmlns:android="http://schemas.android.com/apk/res/android">
    <external-path name="external_files" path="." />
</paths>


PS: @andreaCremisini Sorry couldn't get back to you.

**Comment by awebi** [comment]
@AmithRamesh  i get this error - Error: Unsupported type 'external-files-path'

**Comment by mahavirvataliya** [comment]
i Got error in opening camera

**Comment by mahavirvataliya** [comment]
Error: java.lang.NullPointerException: Attempt to invoke virtual method 'android.content.res.XmlResourceParser android.content.pm.PackageItemInfo.loadXmlMetaData(android.content.pm.PackageManager, java.lang.String)' on a null object reference
